detox/index.d.ts

```diff
@@ -1365,12 +1365,13 @@ declare global {
             setColumnToValue(column: number, value: string): Promise<void>;
 
             /**
-             * Sets the date of a date picker to a date generated from the provided string and date format. (iOS only)
+             * Sets the date of a date picker to a date generated from the provided string and date format.
              * @param dateString string representing a date in the supplied `dateFormat`
              * @param dateFormat format for the `dateString` supplied
              * @example
              * await expect(element(by.id('datePicker'))).toBeVisible();
-             * await element(by.id('datePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "yyyy-MM-dd'T'HH:mm:ssZZZZZ");
+             * await element(by.id('datePicker')).setDatePickerDate('2019-02-06T05:10:00-08:00', "yyyy-MM-dd'T'HH:mm:ssZZZZZ"); //iOS example
```

Comment:
@mauricedoepke I wonder what example we could recommend that would work identically for both the platforms?

Reply:
How about this:
```
/**
  * Sets the date of a date picker to a date generated from the provided string and date format.
  * @param dateString string representing a date in the supplied `dateFormat`
  * @param dateFormat format for the `dateString` supplied
  * @example
  * const datePicker = device.getPlatform() === "ios" ? by.id('datePicker') : by.type('android.widget.DatePicker');
  * await element(datePicker).setDatePickerDate('2019-02-06T05:10:00-08:00', "ISO8601");
  */
```

This works both on android and ios. 

This still contains a case differentiation as @react-native-community/datetimepicker does not support testID's on Android yet. I will try to add get a pr into @react-native-community/datetimepicker that adds it thought. Then we can revisit this example here and unify it completely.

Reply:
I just started a PR on to enable testID's there. If I can get this merged, we can completely unify this example https://github.com/react-native-datetimepicker/datetimepicker/pull/705

Reply:
Hmm, I don't think that the way of getting a `datePicker` element matters here.
I mean that you should put ISO 8601 example first and maybe use a less verbose example for iOS like in the docs.

```js
await element(by.id('datePicker')).setDatePickerDate(new Date(2023, 0, 1).toISOString(), 'ISO8601');
await element(by.id('datePicker')).setDatePickerDate('2023/01/01', "yyyy/MM/dd"); // iOS only
```

Reply:
Thanks, I changed it.

Reply:
@noomorph I adjusted the docs a bit, because a custom date format is now supported on android as well. You may want to check my changes.